shared/gh/js/utils/gh.utils.templates.js

```diff
@@ -18,91 +18,115 @@ define(['exports', 'gh.constants'], function(exports, constants) {
+                    _.cachePartial(name, template);
+
+                    // Return or render the partial HTML
+                    return resolvePartial();
+                });
+
+                // Return the promise
+                return dfd.promise();
             }
         });
 
-        // The collection of templates
-        var deps = [
-            'text!/shared/gh/partials/admin-batch-edit.html',
-            'text!/shared/gh/partials/admin-batch-edit-actions.html',
-            'text!/shared/gh/partials/admin-batch-edit-date.html',
-            'text!/shared/gh/partials/admin-batch-edit-event-row.html',
-            'text!/shared/gh/partials/admin-batch-edit-event-type.html',
-            'text!/shared/gh/partials/admin-batch-edit-time-picker.html',
-            'text!/shared/gh/partials/admin-borrow-series-module-item.html',
-            'text!/shared/gh/partials/admin-edit-date-field.html',
-            'text!/shared/gh/partials/admin-edit-dates.html',
-            'text!/shared/gh/partials/admin-header-template.html',
-            'text!/shared/gh/partials/admin-module-item.html',
-            'text!/shared/gh/partials/admin-modules.html',
-            'text!/shared/gh/partials/agenda-view.html',
-            'text!/shared/gh/partials/borrow-series-modal.html',
-            'text!/shared/gh/partials/calendar.html',
-            'text!/shared/gh/partials/delete-module-modal.html',
-            'text!/shared/gh/partials/delete-module-overview.html',
-            'text!/shared/gh/partials/delete-series-modal.html',
-            'text!/shared/gh/partials/editable-parts.html',
-            'text!/shared/gh/partials/empty-timetable.html',
-            'text!/shared/gh/partials/event.html',
-            'text!/shared/gh/partials/event-popover.html',
-            'text!/shared/gh/partials/login-form.html',
-            'text!/shared/gh/partials/login-modal.html',
-            'text!/shared/gh/partials/header-template.html',
-            'text!/shared/gh/partials/new-module-modal.html',
-            'text!/shared/gh/partials/new-series.html',
-            'text!/shared/gh/partials/rename-module-modal.html',
-            'text!/shared/gh/partials/series-borrowed-popover.html',
-            'text!/shared/gh/partials/series-borrowed-published-popover.html',
-            'text!/shared/gh/partials/series-info.html',
-            'text!/shared/gh/partials/series-info-modal.html',
-            'text!/shared/gh/partials/student-module-item.html',
-            'text!/shared/gh/partials/student-modules.html',
-            'text!/shared/gh/partials/subheader-part.html',
-            'text!/shared/gh/partials/subheader-picker.html',
-            'text!/shared/gh/partials/subheader-pickers.html',
-            'text!/shared/gh/partials/visibility-button.html',
-            'text!/shared/gh/partials/visibility-modal.html'
+        // Collection of partials that can't be conveniently lazy-loaded
+        var preLoadPartials = [
```

Comment:
RequireJS doesn't expose the path of the required file in the callback and we need it to get the corresponding partial name to be able to cache it. We could optimise this a bit by using the constants though. What about just changing this to:

```javascript
var preLoadPartials = [
    'text!' + constants.templates['event']
];
```